Telemetry/AlarmsTest.cs

```diff
@@ -0,0 +1,296 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using Helpers;
+using Helpers.Http;
+using Helpers.Models.TelemetryAlarms;
+using Helpers.Models.TelemetryRules;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Telemetry
+{
+    [Collection("Telemetry Tests")]
+    public class AlarmsTest : IDisposable
+    {
+        private readonly IHttpClient httpClient;
+        private ITestOutputHelper logger;
+
+        private const string DEFAULT_CHILLERS_GROUP_ID = "default_Chillers";
+        private string instantChillerRuleId;
+        private string averageChillerRuleId;
+        private const int ALARM_TRIGGER_WAIT_MSEC = 15000;
+        private const int ALARM_CHECK_RETRY_COUNT = 16;
+
+        private const string ALARMS_ENDPOINT_SUFFIX = "/alarms";
+        private const string RULES_ENDPOINT_SUFFIX = "/rules";
+        private const string ALARMSBYRULE_ENDPOINT_SUFFIX = "/alarmsbyrule";
+
+        // list of rules to delete when tests are complete
+        private List<string> disposeRulesList;
+
+        public AlarmsTest(ITestOutputHelper logger)
+        {
+            this.httpClient = new HttpClient();
+            this.logger = logger;
+            this.disposeRulesList = new List<string>();
+            this.instantChillerRuleId = "INSTANT_CHILLER_TEST_RULE";
+            this.averageChillerRuleId = "AVERAGE_CHILLER_TEST_RULE";
+
+            // Wait for seed data to run simulation before checking alarms
+            Assert.True(SeedData.WaitForSeedComplete());
+
+            // Create rules guaranteed to generate alarms with seed data.
+            this.CreateRuleWithCalculation("Average", "60000", ref this.averageChillerRuleId, false);
+            this.CreateRuleWithCalculation("Instant", "0", ref this.instantChillerRuleId, true);
+        }
+
+        /// <summary>
+        /// Integration test using a real HTTP instance.
+        /// Test that the service starts normally and returns ok status
+        /// </summary>
+        [Fact, Trait(Constants.TEST, Constants.INTEGRATION_TEST)]
+        public void GetListReturnsAlarms_ForAllDevicesInTheLastHour()
+        {
+            // Arrange 
+            const string PARAMETERS = "?devices=chiller-01.0%2C" +
```

Comment:
nit: should we not using hardcode device ids here? Maybe we can improve it later.